apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PatientClaimsEobTaskTransformerV2Test.java

```diff
@@ -0,0 +1,580 @@
+            .get();
+
+    testSnfClaim =
+        parsedRecords.stream()
+            .filter(r -> r instanceof SNFClaim)
+            .map(SNFClaim.class::cast)
+            .findFirst()
+            .get();
+  }
+
+  /**
+   * Verify that the {@link PatientClaimsEobTaskTransformerV2} sucessfully transforms a {@link
+   * CarrierClaim} entity into a {@link ExplanationOfBenefit} FHIR resource and performs no SAMHSA
+   * EOB filtering.
+   */
+  @Test
+  void testTaskTransformerUsingCarrierClaimNoSamhsaFilter() throws IOException {
+    CriteriaQuery<CarrierClaim> clmMockCriteria = mock(CriteriaQuery.class);
+    Root<CarrierClaim> clmRoot = mock(Root.class);
+    setupClaimEntity(mockEntityManager, ClaimTypeV2.CARRIER, clmMockCriteria, clmRoot);
+
+    ClaimTransformerInterfaceV2 claimTransformer =
+        new CarrierClaimTransformerV2(metricRegistry, mockDrugDisplayLookup, mockNpiOrgLookup);
+    PatientClaimsEobTaskTransformerV2 taskTransformer =
+        new PatientClaimsEobTaskTransformerV2(
+            metricRegistry, mockSamhsaMatcher, mockDrugDisplayLookup, mockNpiOrgLookup);
+
+    taskTransformer.setIncludeTaxNumbers(true);
+    taskTransformer.setupTaskParams(
+        claimTransformer, ClaimTypeV2.CARRIER, 1234L, Optional.empty(), Optional.empty(), false);
+    taskTransformer.setEntityManager(mockEntityManager);
+
+    PatientClaimsEobTaskTransformerV2 rslt = taskTransformer.call();
+    assertNotNull(rslt);
+    assertTrue(taskTransformer.ranSuccessfully());
+    assertEquals(1, taskTransformer.fetchEOBs().size());
+    assertEquals(0, taskTransformer.eobsRemovedBySamhsaFilter());
+    assertEquals(-1, taskTransformer.eobsIgnoredBySamhsaFilter());
+  }
+
+  /**
+   * Verify that the {@link PatientClaimsEobTaskTransformerV2} sucessfully transforms a {@link
+   * CarrierClaim} entity into a {@link ExplanationOfBenefit} FHIR resource and performs SAMHSA EOB
+   * filtering.
+   */
+  @Test
+  void testTaskTransformerUsingCarrierClaimWithSamhsa() throws IOException {
+    CriteriaQuery<CarrierClaim> clmMockCriteria = mock(CriteriaQuery.class);
+    Root<CarrierClaim> clmRoot = mock(Root.class);
+    setupClaimEntity(mockEntityManager, ClaimTypeV2.CARRIER, clmMockCriteria, clmRoot);
+
+    ClaimTransformerInterfaceV2 claimTransformer =
+        new CarrierClaimTransformerV2(metricRegistry, mockDrugDisplayLookup, mockNpiOrgLookup);
+    PatientClaimsEobTaskTransformerV2 taskTransformer =
+        new PatientClaimsEobTaskTransformerV2(
+            metricRegistry, mockSamhsaMatcher, mockDrugDisplayLookup, mockNpiOrgLookup);
+
+    taskTransformer.setIncludeTaxNumbers(true);
+    taskTransformer.setupTaskParams(
+        claimTransformer, ClaimTypeV2.CARRIER, 1234L, Optional.empty(), Optional.empty(), true);
+    taskTransformer.setEntityManager(mockEntityManager);
+
+    PatientClaimsEobTaskTransformerV2 rslt = taskTransformer.call();
+    assertNotNull(rslt);
+    assertTrue(taskTransformer.ranSuccessfully());
```

Comment:
And similarly a verify that the samhsa verifier mock was called should be on this one.

Reply:
done.